Basic TanStack Start examples fail with "npm run build / vinxi start"

### Which project does this relate to?

Start

### Describe the bug

The basic example in TanStack Start docs works in "npm run dev", and "npm run build", but the "npm run start" command fails.

I also checked this starter template and it also exhibits the same issue:

https://github.com/dotnize/tanstarter/issues/2

The error I am getting on MacOS ARM64 latest with Node 23.3:

```
% git clone git@github.com:dotnize/tanstarter.git
% cd tanstarter
% npm install
% npm run build
% npm run start

[2:45:00 p.m.]  ERROR  Cannot find module '/Users/benhouston/Coding/Personal/tanstarter/.output/server/index.mjs' imported from /Users/benhouston/Coding/Personal/tanstarter/node_modules/vinxi/bin/cli.mjs

  at finalizeResolution (node:internal/modules/esm/resolve:275:11)
  at moduleResolve (node:internal/modules/esm/resolve:860:10)
  at defaultResolve (node:internal/modules/esm/resolve:984:11)
  at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:654:12)
  at #cachedDefaultResolve (node:internal/modules/esm/loader:603:25)
  at ModuleLoader.resolve (node:internal/modules/esm/loader:586:38)
  at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:242:38)
  at onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:546:36)
  at TracingChannel.tracePromise (node:diagnostics_channel:344:14)
  at ModuleLoader.import (node:internal/modules/esm/loader:545:21) 
```

I also can reproduce on this repository as well that I based on of following the getting started tutorial: https://github.com/bhouston/tanstack-start

I suspect the error may be related to vinxi configuration?

### Your Example Website or App

https://github.com/dotnize/tanstarter/

### Steps to Reproduce the Bug or Issue

On MacOS with ARM64 and Node 23.3.0:

1. git clone git@github.com:dotnize/tanstarter.git
2. cd tanstarter
3. npm install
4. npm run build
5. npm run start - this will fail with the above error message.

### Expected behavior

I would expect it to start the application correctly.

### Screenshots or Videos

_No response_

### Platform

- OS: macOS
- Browser: Node
- Version: [e.g. 91.1]


### Additional context

I am new to TanStack, just learnt it today.  So maybe I am doing something wrong, but it given it also happens on that template repo, I think the issue isn't me.

--- a/packages/start/src/config/index.ts
+++ b/packages/start/src/config/index.ts
@@ -97,7 +97,7 @@ export function defineConfig(
     serverSchema.parse(opts.server || {})
 
   const deploymentPreset = checkDeploymentPresetInput(
-    configDeploymentPreset || 'vercel',
+    configDeploymentPreset || 'node-server',
   )
   const tsr = setTsrDefaults(opts.tsr)
   const appDirectory = tsr.appDirectory

